test/issue-254.test.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const FindMyWay = require('..')
+
+test('Constraints should not be overrided when multiple router is created', t => {
+  t.plan(1)
+
+  let count = 0
+
+  const constraint = {
+    name: 'secret',
+    storage: function () {
+      let secrets = {}
+      return {
+        get: (secret) => { return secrets[secret] || null },
+        set: (secret, store) => { secrets[secret] = store },
+        del: (secret) => { delete secrets[secret] },
+        empty: () => { secrets = {} }
+      }
+    },
+    deriveConstraint: (req, ctx) => {
+      return req.headers['x-secret']
+    },
+    validate () { return true }
+  }
+
+  const router1 = FindMyWay({ constraints: { secret: constraint } })
+  FindMyWay()
+
+  router1.on('GET', '/', { constraints: { secret: 'alpha' } }, () => {})
+  count++
+
+  router1.find('GET', '/', { constraints: { secret: 'alpha' } })
+  count++
+
+  t.equal(count, 2, 'constraints is not overrided')
```

Comment:
I don't get how this test guarantees the constraints are not overridden. Maybe you forgot to check if `router1.find` returns !== null?

Reply:
It must throw an error if the constraints is override. Because there is no secret constraints inside the Constrainer anymore.

Reply:
More breakdown.
1. adding `.on` is check by the `Router` options, so it will not be overrided.
2. using `.find` is the key point. The `constraints` function is used at this point, so it would throw an error if the `constraints` is override.

If the test catch the throwing error, that means the `constraints` is being overrided in `HandlerStorage`.

Reply:
so, maybe is better to change `t.equal` to `t.pass`